File: src/main/java/net/moltendorf/Bukkit/LinkedWhitelist/Listeners.java
package net.moltendorf.Bukkit.LinkedWhitelist;

import net.moltendorf.Bukkit.LinkedWhitelist.storage.StorageException;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.AsyncPlayerPreLoginEvent;

import java.util.UUID;

/**
 * Created by moltendorf on 2015-08-01.
 *
 * @author moltendorf
 */
public class Listeners implements Listener {
	@EventHandler
	public void AsyncPlayerPreLoginEventHandler(final AsyncPlayerPreLoginEvent event) {
		final Settings settings = Settings.getInstance();

		if (settings.isWhitelistEnabled()) {
			final UUID id = event.getUniqueId();
			final String name = event.getName();

			try {
				Boolean permission = settings.getStorage().getPermissionForPlayer(id, name);

				if (permission == null || permission.equals(false)) {
					event.disallow(AsyncPlayerPreLoginEvent.Result.KICK_WHITELIST, "You are not whitelisted.");
				}
			} catch (final StorageException exception) {
				event.disallow(AsyncPlayerPreLoginEvent.Result.KICK_WHITELIST, "An exception occurred while trying to check if you are on the " +
					"whitelist. Please try again and notify an administrator.");
			}
		}
	}
}

File: src/main/kotlin/net/moltendorf/bukkit/linkedwhitelist/Listeners.kt
package net.moltendorf.bukkit.linkedwhitelist

import net.moltendorf.bukkit.linkedwhitelist.storage.StorageException
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.AsyncPlayerPreLoginEvent

/**
 * Created by moltendorf on 2015-08-01.

 * @author moltendorf
 */
class Listeners : Listener {
  @EventHandler
  fun AsyncPlayerPreLoginEventHandler(event: AsyncPlayerPreLoginEvent) {
    val settings = Settings.instance

    if (settings.isWhitelistEnabled) {
      val id = event.uniqueId
      val name = event.name

      try {
        val permission = settings.storage!!.getPermissionForPlayer(id, name)

        if (permission == null || permission == false) {
          event.disallow(AsyncPlayerPreLoginEvent.Result.KICK_WHITELIST, "You are not whitelisted.")
        }
      } catch (exception: StorageException) {
        event.disallow(AsyncPlayerPreLoginEvent.Result.KICK_WHITELIST, "An exception occurred while trying to check if you are on the " + "whitelist. Please try again and notify an administrator.")
      }

    }
  }
}
